### PR TITLE
Switch ESS region to azure-eastus2 for integ tests

### PR DESCRIPTION
**Pull request by pchila**
## Checklist

- ~~[ ] My code follows the style guidelines of this project~~
- ~~[ ] I have commented my code, particularly in hard-to-understand areas~~
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- ~~[ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)~~
- ~~[ ] I have added an integration test or an E2E test~~

## Author's Checklist

- [ ]

## Related issues

- 

## Questions to ask yourself

- How are we going to support this in production? 
- How are we going to measure its adoption? 
- How are we going to debug this?
- What are the metrics I should take care of?
- ...

### CODE REVIEW

File: .buildkite/scripts/steps/integration_tests.sh [hunk 1]
```diff
@@ -8,7 +8,7 @@ DEV=true EXTERNAL=true SNAPSHOT=true PLATFORMS=linux/amd64,linux/arm64 PACKAGES=
 
 # Run integration tests
 set +e
-TEST_INTEG_CLEAN_ON_EXIT=true SNAPSHOT=true mage integration:test
+TEST_INTEG_AUTH_ESS_REGION=azure-eastus2 TEST_INTEG_CLEAN_ON_EXIT=true SNAPSHOT=true mage integration:test
```
Comment (ycombinator) [inline .buildkite/scripts/steps/integration_tests.sh:11]: Could we add an issue to revert this change and maybe add that issue to the next sprint or something, just so we don't forget about it?  Otherwise we'll keep going across data centers between the OGC-provisioned test runner hosts in GCP and ESS deployments in Azure.

And then maybe add a comment here in the code linking to that issue, again, to serve as a reminder?

Comment (pchila) [reply .buildkite/scripts/steps/integration_tests.sh:11]: https://github.com/elastic/elastic-agent/issues/3444

**Comment by ycombinator** [comment]
I think we need to revert this PR and make some fixes to the ESS deployment creation code.  Buildkite is failing with this error:

```
Error: error running test: ogc-linux-arm64-ubuntu-2204-db37 unable to continue because stack never became ready: failed to create ESS cloud for version 8.10.2-SNAPSHOT: failed to create: (cluster.invalid_deployment_template) The referenced deployment template ID [gcp-storage-optimized-v5] does not exist
```

The `gcp-storage-optimized-v5` deployment template cannot be used in Azure. 🤦

**Comment by pchila** [comment]
@ycombinator I see the error on 8.10 backport but main seems fine... are we missing some commits for the integration test framework ?

**Comment by ycombinator** [comment]
> @ycombinator I see the error on 8.10 backport but main seems fine... are we missing some commits for the integration test framework ?

If `main` is fine but `8.10` is broken then, yes, it looks like https://github.com/elastic/elastic-agent/pull/3365 was missing the backport to `8.10`.  I see that got taken care of now and the backport for this PR here also got merged successfully.  Thanks1
